app/src/features/surveys/view/survey-animals/form-sections/MeasurementAnimalForm.tsx

```diff
@@ -0,0 +1,188 @@
+ *
+ * Returns {*}
+ */
+
+const MeasurementAnimalForm = () => {
+  const api = useCritterbaseApi();
+  const { values } = useFormikContext<IAnimal>();
+
+  const { data: measurements, load } = useDataLoader(api.lookup.getTaxonMeasurements);
+
+  if (values.general.taxon_id) {
+    load(values.general.taxon_id);
+  }
+
+  const newMeasurement: IAnimalMeasurement = {
+    _id: v4(),
+
+    taxon_measurement_id: '',
+    value: undefined,
+    qualitative_option_id: '',
+    measured_timestamp: '' as unknown as Date,
+    measurement_comment: undefined
+  };
+
+  return (
+    <FieldArray name={NAME}>
+      {({ remove, push }: FieldArrayRenderProps) => (
+        <>
+          <FormSectionWrapper
+            title={SurveyAnimalsI18N.animalMeasurementTitle}
+            addedSectionTitle={SurveyAnimalsI18N.animalMeasurementTitle2}
+            titleHelp={SurveyAnimalsI18N.animalMeasurementHelp}
+            btnLabel={SurveyAnimalsI18N.animalMeasurementAddBtn}
+            disableAddBtn={!lastAnimalValueValid('measurements', values)}
+            handleAddSection={() => push(newMeasurement)}
+            handleRemoveSection={remove}>
+            {values.measurements.map((measurement, index) => (
+              <MeasurementFormContent key={measurement._id} index={index} measurements={measurements} />
+            ))}
+          </FormSectionWrapper>
+        </>
+      )}
+    </FieldArray>
+  );
+};
+
+interface MeasurementFormContentProps {
+  index: number;
+  measurements?: IMeasurementStub[];
+}
+
+const MeasurementFormContent = ({ index, measurements }: MeasurementFormContentProps) => {
+  const { values, setFieldValue } = useFormikContext<IAnimal>();
+  const [measurement, setMeasurement] = useState<IMeasurementStub>();
+
+  const taxonMeasurementId = values.measurements[index].taxon_measurement_id;
+  const isQuantMeasurement = has(measurement, 'unit');
+
+  const tmIDName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'taxon_measurement_id', index);
+  const valName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'value', index);
+  const opIDName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'qualitative_option_id', index);
+
+  useEffect(() => {
+    setFieldValue(valName, undefined);
+    setFieldValue(opIDName, '');
+    const m = measurements?.find((m) => m.taxon_measurement_id === taxonMeasurementId);
+    setMeasurement(m);
+  }, [taxonMeasurementId]);
+
+  const validateValue = async (val: '' | number) => {
+    const min = measurement?.min_value ?? 0;
+    const max = measurement?.max_value;
+    const unit = measurement?.unit ? ` ${measurement.unit}'s` : ``;
+    if (val === '') {
+      return;
+    }
+    if (val < min) {
+      return `Measurement must be greater than ${min}${unit}`;
+    }
+    if (max && val > max) {
+      return `Measurement must be less than ${max}${unit}`;
+    }
+  };
+
+  return (
+    <Fragment key={`meausurement-inputs-${index}`}>
```

Comment:
nit: spelling

Reply:
Removed the key here. Parent rendering list of components above those this isnt needed.